kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -927,6 +949,16 @@ public void givenBasicMockedDbusConnector() throws DBusException, IOException {
         when(this.dbusConnection.getRemoteObject(eq("org.freedesktop.ModemManager1"),
                 eq("/org/freedesktop/ModemManager1"), any()))
                 .thenReturn(this.mockedModemManager);
+        
+        when(this.dbusConnection.getRemoteObject(eq("fi.w1.wpa_supplicant1"),
+                eq("/fi/w1/wpa_supplicant1"), any())).thenReturn(this.mockedWpaSupplicant);
+        
+        DBusPath fakeDbusInterfacePath = new DBusPath("/fake/dbus/path");
+        
+        when(this.mockedWpaSupplicant.GetInterface(anyString())).thenReturn(fakeDbusInterfacePath);
+        
+        when(this.dbusConnection.getRemoteObject(eq("fi.w1.wpa_supplicant1"), eq(fakeDbusInterfacePath.getPath()), any()))
+        .thenReturn(this.mockedInterface);
```

Comment:
This is not the correct way to mock the device. You should create and return the dbus path only for Wifi Devices and only for the interfaceId you expect.

```suggestion
        
        when(this.dbusConnection.getRemoteObject(eq("fi.w1.wpa_supplicant1"),
                eq("/fi/w1/wpa_supplicant1"), any())).thenReturn(this.mockedWpaSupplicant);
        
```

You should mock the interface in the [`givenMockedDevice` method](https://github.com/eclipse/kura/blob/177821520ceb573539fd6f0aad461990a98d0ec5/kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java#L986) where you can mock the interface only for WiFi devices:

https://github.com/eclipse/kura/blob/177821520ceb573539fd6f0aad461990a98d0ec5/kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java#L1017

There you'll have:

```java
    if (type == NMDeviceType.NM_DEVICE_TYPE_WIFI) {

        DBusPath fakeDbusInterfacePath = new DBusPath("/fake/dbus/path");
        
        when(this.mockedWpaSupplicant.GetInterface(interfaceId)).thenReturn(fakeDbusInterfacePath);
        
        when(this.dbusConnection.getRemoteObject(eq("fi.w1.wpa_supplicant1"), eq(fakeDbusInterfacePath.getPath()), any()))
        .thenReturn(this.mockedInterface);

         simulateIwCommandOutputs(interfaceId, mockedProperties1);
}
```